### PR TITLE
Expand the "notification" addressee list for ADOPTME etc

### LINKED ISSUE #2268: HANDOFF banner should be more verbose
When HANDOFF has permission for a distribution/module, this banner text is added:

> Take me over?
> The maintainer of this distribution is looking for someone to take over! If you're interested then please contact them via email.

Could that mailto: link also include the address of the last maintainer (if different from the first-come permission holder) -- or even include all the comaint holders in cc? I've had a few distributions taken over without my knowledge where I released it last but only have comaint rather than first-come, and I'd have been happy to continue maintaining the distribution (and incorporating the new arrival's patches) if only I had known.

I would also include explicit language in the body of the email noting what user did the last release, and the date of that release -- as often the first-come holder has lost track of what has happened with the distribution since his last release and other people took over, and this saves him the effort of having to look that up. Also it would be good to linkify the module name rather than just including it in plain text.

How about:

    - Hey $first_come_name,
    -
    - I'm interested in taking over the $module_name module. If you can grant me the privilege I will release the next version.

    + Hey $first_come_name (and co-maintainers $name, $name, $name),
    +
    + I would like to take over the <a href="https://metacpan.org/pod/$module_name">$module_name</a> module, last
    released on $date by $releaser_name. If you can grant me the privilege I will release the next version.

This would at least let other comaintainers have a chance to say "wait a minute!".

### LINKED ISSUE #2293: HANDOFF gives email to last releaser, not necessarily first-come
I saw this on `Event` earlier. The then first-come, JPRIT, had added HANDOFF as a co-maint. The email link was to the last releaser (me), who couldn't do anything about it, wasting the emailer's time.

Comments / caveats:
- This isn't visible now, since then I've become first-come and removed `HANDOFF`.
- The HANDOFF-adding happened without me knowing, which is probably a missing PAUSE feature, not a MetaCPAN matter. It seems like there should be a pushed notification.

 #2268 says the banner isn't verbose enough, but by my reading it was Ok. I think (similar to that issue) the addresses mailed to should be `To` the first-come, `Cc` at least the last releaser, and probably all co-maints.

### PR DESCRIPTION
**Pull request by mohawk2**
~~This suppresses `writev` warnings on `EPIPE`.~~ But mainly it addresses, and therefore hopefully fix #2268 and fix #2293. @karenetheridge Do you agree?

EDIT: remove the `writev` stuff in favour of #2346

### CODE REVIEW

**Comment by karenetheridge** [comment]
Am I inferring correctly that this PR changes the people in the mailto: link to everyone who has comaint, not just the first-come holder?

**Comment by mohawk2** [comment]
@karenetheridge Yes, that's what I was trying to achieve?

File: lib/MetaCPAN/Web/Model/API/Permission.pm [hunk 1]
```diff
@@ -89,6 +89,7 @@ sub _permissions_to_notification {
     Future->done( {
         took         => $data->{took} || 0,
         notification => $type,
+        notifiable   => [ grep !$special{$_}, @perm_holders ],
```
Comment (haarg) [inline lib/MetaCPAN/Web/Model/API/Permission.pm:92]: This shouldn't be a separate item.  The `notification` value should be changed to a hash with type, email, and module elements.

File: lib/MetaCPAN/Web/Model/ReleaseInfo.pm [hunk 2]
```diff
@@ -133,37 +133,54 @@ sub normalize {
...
-                    modules => $data->{modules}{modules},
-                    changes => $data->{changes}{changes},
-                    )
-                : ()
-            ),
+        my $n_data = $data->{notification};
+        my $n_future;
+        if ( $n_data->{notification} ) {
+            $n_future
+                = $self->_author->get_multiple( @{ $n_data->{notifiable} } )
```
Comment (haarg) [inline lib/MetaCPAN/Web/Model/ReleaseInfo.pm:140]: Any logic like this should move into the API::Permissions get_notification_type method.

File: root/inc/notification.html [hunk 3]
```diff
@@ -4,38 +4,36 @@
...
-            body   = format("Hey %s,\r\n\r\nI'm interested in taking over the %s module. If you can grant me the privilege I will release the next version.\r\n\r\n Kind Regards,")
+            body   = format("Hello,\r\n\r\nI'm interested in taking over the %s module. If you can grant me the privilege I will release the next version.\r\n\r\n Kind Regards,")
         },
     }.$notification;
+    IF NOTIFICATION.email;
+        notifiable.push(NOTIFICATION.email);
+    END;
     IF pod_file && pod_file.module.0;
-        NOTIFICATION.module = pod_file.module.0.name;
+        NOTIFICATION.module = pod_file.module.0.name || release.main_module;
```
Comment (haarg) [inline root/inc/notification.html:28]: This should be pulling from the notification hash, as described in my other comment.

Comment (mohawk2) [reply root/inc/notification.html]: Now implemented

File: lib/MetaCPAN/Web/Model/API/Permission.pm (hunk 1 above)
Comment (mohawk2) [reply lib/MetaCPAN/Web/Model/API/Permission.pm]: This is done.

File: lib/MetaCPAN/Web/Model/ReleaseInfo.pm (hunk 2 above)
Comment (mohawk2) [reply lib/MetaCPAN/Web/Model/ReleaseInfo.pm]: Now implemented, and ready for re-review!

**Comment by mohawk2** [comment]
`t/model/permission.t` now provides a pattern for how to mock the API. This could probably be generalised a bit more.

**Pull request merged**